core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt

```diff
@@ -151,3 +179,137 @@ class OnlineMultiplayer {
         return GameSaver.gameInfoPreviewFromString(Gzip.unzip(zippedGameInfo))
     }
 }
+
+/**
+ * Used to communicate data access between players
+ */
+class LockFile {
+    // The lockData is necessary to make every LockFile unique
+    // If Dropbox gets a file with the same content and overwrite set to false, it returns no
+    // error even though the file was not uploaded as the exact file is already existing
+    var lockData = UUID.randomUUID().toString()
+}
+
+/**
+ *	Wrapper around OnlineMultiplayer's synchronization facilities.
+ *
+ *	Based on the design of Mutex from kotlinx.coroutines.sync, except that when it blocks,
+ *	  it blocks the entire thread for an increasing period of time via Thread.sleep()
+ */
+class ServerMutex(val gameInfo: GameInfoPreview) {
+    private var locked = false
+
+    constructor(gameInfo: GameInfo) : this(gameInfo.asPreview()) { }
+
+    /**
+     * Try to obtain the server lock ONCE
+     * DO NOT forget to unlock it when you're done with it!
+     * Sleep between successive attempts
+     * @see lock
+     * @see unlock
+     * @return true if lock is acquired
+     */
+    fun tryLock(): Boolean {
+        // If we already hold the lock, return without doing anything
+        if (locked) {
+            return locked
+        }
+        
+        locked = false
+        val fileName = "${OnlineMultiplayer().getGameLocation(gameInfo.gameId)}_Lock"
+
+        // We have to check if the lock file already exists before we try to upload a new
+        // lock file to not overuse the dropbox file upload limit else it will return an error
+        try {
+            val stream = DropBox.getFileMetaData(fileName)
+            val reader = BufferedReader(InputStreamReader(stream))
+            val metaData = GameSaver.json().fromJson(DropboxMetaData::class.java, reader.readText())
+
+            val date = metaData?.getServerModified()
+            println(date?.time)
+            println(System.currentTimeMillis())
+            // 30 seconds should be more than sufficient for everything lock related
+            // so we can assume the lock file was forgotten if it is older than 30 sec
+            if (date != null && System.currentTimeMillis() - date.time < 30000) {
+                return locked
+            } else {
+                DropBox.deleteFile(fileName)
+            }
```

Comment:
I tried to implement the lock release for forgotten lock files but the Date object returned by String.parseDate() returns an epoch timestamp which is offset by one hour (UTC -1) compared to System.currentTimeMillis(). Shouldn't epoch time be timezone independent? Don't get what's going on, so just wanted to ask if someone has a clue or if I just have to check for < 3630000 instead of 30000 ¯\_(ツ)_/¯.